### ISSUE TITLE
Separate setPermission from setting signers

### ISSUE THREAD

**Issue by Ramarti** [storyprotocol/protocol-core-v1#98]
The way of adding a signer (meaning an account that can execute anything on an IPAccount, same as owner) it's part of `setPermissions` in AccessController:

`accessController.setPermission(ipAccount, signer, address(0), bytes4(0), AccessPermission.ALLOW)`

This is dangerous for multiple reasons:
- It could be set by error if the user/sdk/tooling doesn't fill `target` and `func` when calling
- It would be easy for a malicious module to trick users into giving complete admin permissions by sneaking the above between multiple function level permissions. `setBatchPermissions(AccessPermission.Permission[] memory permissions)`
- The UX from block explorers and wallets will make it even harder for the user to distinguish, because it's the same method signature with very very different security profile.

We must remove that level of permission from AccessController, and instead have a in IPAccountImpl

```
function setAllowedSigner(address signer) onlyOwner {
     allowedSigners[signer] = true;
    emit NewSignerAdded(signer);
}
```

This way, existing indexers and explorers can properly flag the action.

in IpAccount, where we can store a simple mapping of signer addresses, then:

```
function _isValidSigner(address signer, address to, bytes calldata data) internal view returns (bool) {
        if (data.length > 0 && data.length < 4) {
            revert Errors.IPAccount__InvalidCalldata();
        }
        bytes4 selector = bytes4(0);
        if (data.length >= 4) {
            selector = bytes4(data[:4]);
        }
        if (msg.sender == owner || allowedSigners[msg.signer]) {
               return true
        }
        // the check will revert if permission is denied
        IAccessController(accessController).checkPermission(address(this), signer, to, selector);
        return true;
    
}
```

**Comment by kingster-will** [comment]
Thank you for raising this concern. I understand the points you've made regarding the potential security issues with the current implementation of setPermission in the AccessController. While it's true that this method could lead to inadvertent permissions being granted if not handled carefully, it's comparable in risk to the `approve()` function seen in the ERC20 protocol, which also requires careful management by users.

In terms of centralized vs. decentralized control management, maintaining all access control mechanisms within a single module can simplify oversight and security auditing. However, your suggestion to improve usability and security by separating the concerns makes sense. To this end, introducing a distinct function like `setAllPermissions()` in AccessController.

**Comment by kingster-will** [comment]
This issue has already been addressed with issue #122.
